kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtNetworkServiceImpl.java

```diff
@@ -1237,252 +1046,477 @@ public ArrayList<GwtFirewallNatEntry> findDeviceFirewallNATs(GwtXSRFToken xsrfTo
+        String passKey = GwtSafeHtmlUtils.htmlUnescape(gwtModemConfig.getPassword());
+        if (passKey != null && passKey.equals(PASSWORD_PLACEHOLDER)) {
+            List<GwtNetInterfaceConfig> result = privateFindNetInterfaceConfigurations();
+            for (GwtNetInterfaceConfig netConfig : result) {
+                if (netConfig instanceof GwtModemInterfaceConfig) {
+                    GwtModemInterfaceConfig oldModemConfig = (GwtModemInterfaceConfig) netConfig;
+                    if (gwtModemConfig.getName().equals(oldModemConfig.getName())) {
+                        properties.put(basePropName + "password",
+                                new Password(GwtSafeHtmlUtils.htmlUnescape(oldModemConfig.getPassword())));
+                    }
+                }
+            }
+        } else if (passKey != null) {
+            properties.put(basePropName + "password", new Password(passKey));
+        }
+    }
+
+    private void fillDhcpAndNatProperties(GwtNetInterfaceConfig config, Map<String, Object> properties,
+            String basePropName) throws KuraException, UnknownHostException {
+        String routerMode = config.getRouterMode();
+        String natEnabledPropName = basePropName + "nat.enabled";
+        StringBuilder dhcpServer4PropName = new StringBuilder(basePropName).append("dhcpServer4.");
+        if (routerMode.equals(GwtNetRouterMode.netRouterOff.name())) {
+            logger.debug("DCHP and NAT are disabled");
+            properties.put(dhcpServer4PropName.toString() + ENABLED, false);
+            properties.put(natEnabledPropName, false);
+        } else if (routerMode.equals(GwtNetRouterMode.netRouterDchp.name())) {
+            logger.debug("DCHP is enabled");
+            fillDhcpServerProperties(config, properties, basePropName);
+            properties.put(natEnabledPropName, false);
+        } else if (routerMode.equals(GwtNetRouterMode.netRouterDchpNat.name())) {
+            logger.debug("DCHP and NAT is enabled");
+            fillDhcpServerProperties(config, properties, basePropName);
+            properties.put(natEnabledPropName, true);
+        } else if (routerMode.equals(GwtNetRouterMode.netRouterNat.name())) {
+            logger.debug("NAT is enabled");
+            properties.put(dhcpServer4PropName.toString() + ENABLED, false);
+            properties.put(natEnabledPropName, true);
+        } else {
+            logger.error("Unsupported routerMode: {}", routerMode);
+            throw new KuraException(KuraErrorCode.CONFIGURATION_ERROR, "Unsupported routerMode: " + routerMode);
+        }
+    }
+
+    private void fillDhcpServerProperties(GwtNetInterfaceConfig config, Map<String, Object> properties,
+            String basePropName) throws UnknownHostException {
+        StringBuilder dhcpServer4PropName = new StringBuilder(basePropName).append("dhcpServer4.");
+        properties.put(dhcpServer4PropName.toString() + ENABLED, true);
```

Comment:
`StringBuilder.toString()` allocates a new string at each invocation. As an enhancement we could refactor the code to avoid calling this method multiple times to produce the same string.